test/unit/overlay.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+var fixtures = require('../fixtures');
+var fs = require('fs');
+var sharp = require('../../index');
+
+sharp.cache(0);
+
+// Main
+describe('Overlays', function() {
+  it('Overlay transparent PNG on solid background', function(done) {
+    sharp(fixtures.inputPngOverlayLayer0)
+      .overlayWith(fixtures.inputPngOverlayLayer1)
+      .toBuffer(function (error, data, info) {
+        if (error) return done(error);
```

Comment:
The mocha [async examples](http://mochajs.org/#asynchronous-code) contain both `throw err` and `done(err)` variants. Reducing the number of conditional branches in test code decreases the need to test-the-tests. In this case, both approaches introduce one `if/else` split. Promises (and generators) will take over the world soon so we can, ahem, *throw* the error handling part of this test code away at some point.